presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java

```diff
@@ -85,7 +89,9 @@
     private final Optional<BlockedReason> blockedReason;
 
     @Nullable
-    private final OperatorInfo info;
+    private OperatorInfo info;
+    @Nullable
+    private OperatorInfoUnion infoUnion;
```

Comment:
Just assign one or the other to null so we can keep these as immutable?

```suggestion
    private final OperatorInfo info;
    @Nullable
    private final OperatorInfoUnion infoUnion;
```

Reply:
as discussed offline, OperatorInfo is used in multiple places and is an interface. The plan is not to deprecate the class immediately.

Reply:
done. Added final modifiers.